sympy/functions/elementary/complexes.py

```diff
@@ -503,6 +503,15 @@ def eval(cls, arg):
             if not unk or not all(conj.has(conjugate(u)) for u in unk):
                 return sqrt(expand_mul(arg*conj))
 
+        # if the expression is a function, then try to see if its nonnegative or negative
+        # Some class of Functions like UndefinedFunctions return a `property` function
+        if isinstance(arg, Function):
+            if arg.is_nonnegative is True or arg.is_positive is True:
```

Comment:
Oh. The reason it didn't work earlier is because `is_nonnegative` and `is_positive` wasn't implemented for `cosh`. 
Still, it didn't work because I only implemented `is_positive` for `cosh`.
I'm removing that case then.